diff --git a/dist/packages/@ember/renderer/index.js b/dist/packages/@ember/renderer/index.js
index 1e7f5c9cd96f38f3ffd0dff6870b7e097a1a0758..ce998b530d888baf8e1409294c3c887844301f95 100644
--- a/dist/packages/@ember/renderer/index.js
+++ b/dist/packages/@ember/renderer/index.js
@@ -8,3 +8,4 @@ import '../../@glimmer/reference/index.js';
 import '../../@glimmer/validator/index.js';
 import '@embroider/macros';
 import '../../@glimmer/manager/index.js';
+export { subtle } from './sync.js';
diff --git a/dist/packages/@ember/renderer/sync.js b/dist/packages/@ember/renderer/sync.js
new file mode 100644
index 0000000000000000000000000000000000000000..a3c477bac23c9ebea3ebdf54669e5938a8e8eaf7
--- /dev/null
+++ b/dist/packages/@ember/renderer/sync.js
@@ -0,0 +1,21 @@
+import { createCache, getValue } from '../../@glimmer/validator/index.js';
+
+let caches = [];
+
+
+export const subtle = {
+  flushSync() {
+    caches.forEach(cache => {
+      getValue(cache);
+    });
+  },
+  sync(fn) {
+    let cache = createCache(fn);
+
+    caches.push(cache);
+
+    return () => {
+      caches = caches.filter(c => c!==cache)
+    }
+  }
+}
diff --git a/dist/packages/shared-chunks/index-D-GTx_Yt.js b/dist/packages/shared-chunks/index-D-GTx_Yt.js
index 274d4d6dd3978811308ecfbcf6c27ccc51e10a36..702c4ca8f70954b13b2b5ee9bbdc22439a93433d 100644
--- a/dist/packages/shared-chunks/index-D-GTx_Yt.js
+++ b/dist/packages/shared-chunks/index-D-GTx_Yt.js
@@ -1,3 +1,4 @@
+import { subtle } from '../@ember/renderer/index.js';
 import { templateFactory, EvaluationContextImpl } from '../@glimmer/opcode-compiler/index.js';
 import { g as getFactoryFor, p as privatize } from './registry-BJpQx6hv.js';
 import { warn, debugFreeze, deprecate } from '../@ember/debug/index.js';
@@ -5470,6 +5471,7 @@ class Renderer {
             continue;
           }
           root.render();
+          subtle.flushSync();
         }
         this._lastRevision = valueForTag(CURRENT_TAG);
       });
