diff --git a/dist/packages/shared-chunks/index-D-GTx_Yt.js b/dist/packages/shared-chunks/index-D-GTx_Yt.js
index 274d4d6dd3978811308ecfbcf6c27ccc51e10a36..e37cb2c8032e5d64886f47109aaeb5c62d2bf237 100644
--- a/dist/packages/shared-chunks/index-D-GTx_Yt.js
+++ b/dist/packages/shared-chunks/index-D-GTx_Yt.js
@@ -5293,8 +5293,15 @@ class Renderer {
     } = props;
     let owner = getOwner(props);
     (isDevelopingApp() && !(owner) && assert('Renderer is unexpectedly missing an owner', owner));
-    let document = owner.lookup('service:-document');
-    let env = owner.lookup('-environment:main');
+    let document = globalThis.document;
+    let env = {
+      hasDOM: true,
+      isInteractive: true,
+      history: null,
+      isChrome: false,
+      isFirefox: false,
+      location: null,
+    };
     let rootTemplate = owner.lookup(privatize`template:-root`);
     let builder = owner.lookup('service:-dom-builder');
     return new this(owner, document, env, rootTemplate, _viewRegistry, builder);
